types/node/v14/fs/promises.d.ts

```diff
@@ -184,10 +184,17 @@ declare module 'fs/promises' {
     /**
      * Asynchronous open(2) - open and possibly create a file.
      * @param path A path to a file. If a URL is provided, it must use the `file:` protocol.
-     * @param mode A file mode. If a string is passed, it is parsed as an octal integer. If not
+     * @param [flags='r'] See `support of file system `flags``.
+     * @param [mode] A file mode. If a string is passed, it is parsed as an octal integer. If not
      * supplied, defaults to `0o666`.
      */
-    function open(path: PathLike, flags: string | number, mode?: Mode): Promise<FileHandle>;
+    function open(path: PathLike, flags: string | number | undefined, mode?: string | number): Promise<FileHandle>;
```

Comment:
I'm missing something - why not just make it optional as in v17?

Reply:
oh, just too many versions, let me do this locally.

Reply:
amended